src/client-agent/start_agent.c

```diff
@@ -132,6 +132,7 @@ bool connect_server(int server_id, bool verbose)
 /* Send synchronization message to the server and wait for the ack */
 void start_agent(int is_startup)
 {
+    agent_setup_hostnames();
```

Comment:
I think we should move this into the is_startup condition to only try to solve the hostnames once.